cypress/tests/overview/settings-cluster-general.cy.ts

```diff
@@ -0,0 +1,96 @@
+      cy.get('[data-test="detail-item-value"]').contains('Up to date');
+    });
+  });
+
+  describe('Test Live migration', () => {
+    it('set live migration limits', () => {
+      cy.contains('General settings').click();
+      cy.contains('Live migration').click();
+      cy.get('input[name="parallelMigrationsPerCluster"]').clear().type('4');
+      cy.get('input[name="parallelOutboundMigrationsPerNode"]').clear().type('1');
+      cy.contains('Set live migration network').click();
+      cy.wait(1000);
+    });
+
+    it('verify live migration limits in HCO', () => {
+      const maxPerCluster = '.spec.liveMigrationConfig.parallelMigrationsPerCluster';
+      const maxPerNode = '.spec.liveMigrationConfig.parallelOutboundMigrationsPerNode';
+      const spec = '.spec.liveMigrationConfig';
+      const cmd = `oc get -n openshift-cnv hyperconverged kubevirt-hyperconverged -o jsonpath='{${spec}}'`;
+      cy.exec(cmd).then((res) => {
+        cy.task('log', res.stdout);
+      });
+      cy.checkHCOSpec(maxPerCluster, '4', true);
+      cy.checkHCOSpec(maxPerNode, '1', true);
+    });
+  });
+
+  describe('Test Memory density', () => {
+    it('ID(CNV-11225) enable memory density', () => {
+      cy.contains('Memory density').click();
+      cy.get('[id="memory-density-feature"]').find('input[type="checkbox"]').check({ force: true });
+    });
+
+    it('verify higherWorkloadDensity in HCO', () => {
+      const percentage = '.spec.higherWorkloadDensity.memoryOvercommitPercentage';
+      cy.checkHCOSpec(percentage, '150', true);
+      cy.wait(10000);
+    });
+
+    it('ID(CNV-11225) disable memory density', () => {
+      cy.get('[id="memory-density-feature"]')
+        .find('input[type="checkbox"]')
+        .uncheck({ force: true });
+    });
+
+    it('verify higherWorkloadDensity in HCO', () => {
+      const percentage = '.spec.higherWorkloadDensity.memoryOvercommitPercentage';
+      cy.checkHCOSpec(percentage, '100', true);
+    });
+  });
+
+  describe('Test Automatic images download', () => {
+    it('ID(CNV-11223) disable centos-stream-9-image-cron', () => {
+      cy.contains('Automatic images download').click();
+      cy.get('#centos-stream9-image-cron-auto-image-download-switch')
+        .find('input.pf-v5-c-switch__input')
+        .uncheck({ force: true });
+      cy.wait(10000);
+    });
+
+    it('delete centos-stream9 volumesnapshot', () => {
+      const cmd =
+        'oc delete volumesnapshot -n openshift-virtualization-os-images -l cdi.kubevirt.io/dataImportCron=centos-stream9-image-cron';
+      cy.exec(cmd);
+      cy.wait(10000);
+    });
+
+    it('enable centos-stream9-image-cron again', () => {
+      cy.get('#centos-stream9-image-cron-auto-image-download-switch')
+        .find('input.pf-v5-c-switch__input')
+        .check({ force: true });
+      cy.wait(60000);
+    });
+
+    it('check centos-stream9 volumesnapshot is downloaded again', () => {
+      cy.wait(60000);
+      const cmd =
```

Comment:
instead of one long timeout we could check the condition in a loop with a smaller timout

Reply:
put the block into retry

Reply:
In this case re-try on the entire test case is OK but in many cases you need it for a single step.
Checkout how @sjd78 implemented active checking in a loop:
https://github.com/konveyor/tackle2-ui/blob/991b02aa1001bf05eb53e868e48b8bb42091e48f/hack/setup-operator.sh#L43
Usage:
```
  echo "Waiting for the Tackle CRD to exist"
  retry_command 10 10 \
    $KUBECTL get customresourcedefinitions.apiextensions.k8s.io tackles.tackle.konveyor.io

```
This way you can provide a big maximal timeout i.e. 10min but at the same time be quick if thing are going smoothly.

Reply:
it runs via cy.exec, the wrapped command is hard to use there. it reports some error based on my experience in the past.